### ISSUE TITLE
Reason: Juju failed to bootstrap: maas

### ISSUE THREAD

**Issue by odkhuu11** [conjure-up/conjure-up#822]
Thank you for trying **conjure-up**! Before reporting a bug please make sure you've gone through this checklist:

- [ ] Is this problem already documented at http://conjure-up.io/docs/en/users/#_common_spell_problems
- [ ] Is conjure-up running inside a virtual machine?

Please outline the hardware you're running conjure-up on, including number of cpus, amount of memory, amount of storage space available:

What spell was selected?:

What cloud provider was selected (this includes both MAAS and localhost)?:

Please outline what commands were run to install and execute conjure-up:

Please post **~/.cache/conjure-up/conjure-up.log**

Provide the output of the following commands:

```
which juju
which conjure-up
which lxd
juju version
conjure-up --version
lxd --version
cat /etc/lsb-release
```

Summary of problem:
My MAAS server: Dell 3020
Select Landscape: Dell 990 (but 16GB RAM, 500 GB HDD)
OS: Ubuntu 16.04
~/.cache/conjure-up/conjure-up.log: 
11:21:43 INFO  juju.environs.tools tools.go:101 finding agent binaries in stream "released"

11:21:43 INFO  juju.environs.tools tools.go:103 reading agent binaries with major.minor version 2.1

11:21:43 INFO  juju.environs.tools tools.go:111 filtering agent binaries by version: 2.1.2

11:21:43 INFO  juju.environs.tools tools.go:114 filtering agent binaries by series: xenial

11:21:43 INFO  juju.environs.tools tools.go:117 filtering agent binaries by architecture: amd64

11:21:43 DEBUG juju.environs.tools urls.go:109 trying datasource "keystone catalog"

11:21:45 DEBUG juju.environs.simplestreams simplestreams.go:680 using default candidate for content id "com.ubuntu.juju:released:tools" are {20161007 mirrors:1.0 content-download streams/v1/cpc-mirrors.sjs$

11:21:48 INFO  juju.environs.bootstrap tools.go:74 found 1 packaged agent binaries

11:21:48 INFO  cmd cmd.go:141 Starting new instance for initial controller

Launching controller instance(s) on maas...

11:21:49 INFO  juju.provider.common destroy.go:20 destroying model "controller"

11:21:49 INFO  juju.provider.common destroy.go:31 destroying instances

11:21:49 INFO  juju.provider.common destroy.go:51 destroying storage

11:21:49 ERROR cmd supercommand.go:458 failed to bootstrap model: cannot start bootstrap instance: cannot run instances: cannot run instance: No available machine matches constraints: mem=3584.0 name=Contr$

11:21:49 DEBUG cmd supercommand.go:459 (error details: [{github.com/juju/juju/cmd/juju/commands/bootstrap.go:574: failed to bootstrap model} {github.com/juju/juju/provider/common/bootstrap.go:179: cannot s$

2017-04-14 11:21:50,936 [DEBUG] log.py:14 - conjure-up/_unspecified_spell: Showing dialog for exception: Juju failed to bootstrap: maas
=======================================================================

odkhuu@maas:~$ which juju
/snap/bin/juju
odkhuu@maas:~$ which conjure-up
/snap/bin/conjure-up
odkhuu@maas:~$ which lxd
/usr/bin/lxd
odkhuu@maas:~$ juju version
2.1.2-xenial-amd64
odkhuu@maas:~$ conjure-up --version
conjure-up 2.1.3
odkhuu@maas:~$ lxd --version
2.12
odkhuu@maas:~$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.2 LTS"
odkhuu@maas:~$
![image](https://cloud.githubusercontent.com/assets/27420888/25034772/e80d5776-2123-11e7-8d47-46d641a4ac8f.png)
Please!  What a problem?

**Comment by battlemidget** [comment]
You're constraints seem to be wrong:

```
11:21:49 ERROR cmd supercommand.go:458 failed to bootstrap model: cannot start bootstrap instance: cannot run instances: cannot run instance: No available machine matches constraints: mem=3584.0 name=Contr$
```

**Comment by khunbish** [comment]
ERROR no API addresses
